project-base/storefront/components/Blocks/OrderSummary/PromoCode.tsx

```diff
@@ -5,28 +5,39 @@ import {
     OrderSummaryRowWrapper,
     OrderSummaryTextAndImage,
 } from './OrderSummaryElements';
-import { TypePrice } from 'graphql/types';
+import { TypePromoCode, TypePromoCodeTypeEnum } from 'graphql/types';
 import useTranslation from 'next-translate/useTranslation';
+import { formatPercent } from 'utils/formaters/formatPercent';
 import { useFormatPrice } from 'utils/formatting/useFormatPrice';
 import { isPriceVisible } from 'utils/mappers/price';
 
 type PromoCodeProps = {
-    promoCode: string;
-    discount: TypePrice;
+    promoCode: TypePromoCode;
 };
 
-export const PromoCode: FC<PromoCodeProps> = ({ discount, promoCode }) => {
-    const { t } = useTranslation();
+const PromoCodeValue: FC<{ value: string; type: TypePromoCodeTypeEnum }> = ({ value, type }) => {
     const formatPrice = useFormatPrice();
 
+    const isNominal = type === TypePromoCodeTypeEnum.Nominal;
+    const isPercent = type === TypePromoCodeTypeEnum.Percent;
+
+    return (
+        <p className="text-sm font-bold">-{isNominal ? formatPrice(value) : isPercent ? formatPercent(value) : null}</p>
+    );
+};
+
+export const PromoCode: FC<PromoCodeProps> = ({ promoCode }) => {
+    const { t } = useTranslation();
+    const { discount, code, type } = promoCode;
+
     return (
         <OrderSummaryRowWrapper>
```

Comment:
Would be nice to simplify these wrappers, but it's nice to have. It's up to you, just saying :)

Reply:
As described in the above comment.